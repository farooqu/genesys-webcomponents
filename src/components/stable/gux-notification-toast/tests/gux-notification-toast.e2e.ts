import { newE2EPage } from '@stencil/core/testing';

describe('gux-notification-toast', () => {
  describe('#render', () => {
    [
      {
        description: 'should render neutral notification toast',
        html: `
          <gux-notification-toast lang="en" accent="neutral">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="title">Title</div>
            <div slot="message">This is the message</div>
          </gux-notification-toast>
        `
      },
      {
        description: 'should render positive notification toast',
        html: `
          <gux-notification-toast lang="en" accent="positive">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="title">Title</div>
            <div slot="message">This is the message</div>
          </gux-notification-toast>
        `
      },
      {
        description: 'should render alert notification toast',
        html: `
          <gux-notification-toast lang="en" accent="alert">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="title">Title</div>
            <div slot="message">This is the message</div>
          </gux-notification-toast>
        `
      },
      {
        description: 'should render warning notification toast',
        html: `
          <gux-notification-toast lang="en" accent="warning">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="title">Title</div>
            <div slot="message">This is the message</div>
          </gux-notification-toast>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newE2EPage({ html });
        const element = await page.find('gux-notification-toast');

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });

  describe('dismiss', () => {
    it('click dismiss button', async () => {
      const html = `
        <gux-notification-toast lang="en" accent="neutral">
          <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
          <div slot="title">Title</div>
          <div slot="message">This is the message</div>
        </gux-notification-toast>
      `;
      const page = await newE2EPage({ html });
      const dismissButton = await page.find('.gux-dismiss-button');
      const guxdismissSpy = await page.spyOnEvent('guxdismiss');
      const clickSpy = await page.spyOnEvent('click');

      expect(guxdismissSpy).not.toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-notification-toast')).not.toBeNull();

      await dismissButton.click();
      await page.waitForChanges();

      expect(guxdismissSpy).toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-notification-toast')).toBeNull();
    });

    it('click dismiss button and prevent default', async () => {
      const html = `
        <gux-notification-toast lang="en" accent="neutral">
          <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
          <div slot="title">Title</div>
          <div slot="message">This is the message</div>
        </gux-notification-toast>
      `;
      const page = await newE2EPage({ html });
      const dismissButton = await page.find('.gux-dismiss-button');
      const guxdismissSpy = await page.spyOnEvent('guxdismiss');
      const clickSpy = await page.spyOnEvent('click');

      await page.evaluate(() => {
        document.addEventListener('guxdismiss', event => {
          event.preventDefault();
        });
      });

      expect(guxdismissSpy).not.toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-notification-toast')).not.toBeNull();

      await dismissButton.click();
      await page.waitForChanges();

      expect(guxdismissSpy).toHaveReceivedEvent();
      expect(clickSpy).not.toHaveReceivedEvent();
      expect(await page.find('gux-notification-toast')).not.toBeNull();
    });
  });
});
