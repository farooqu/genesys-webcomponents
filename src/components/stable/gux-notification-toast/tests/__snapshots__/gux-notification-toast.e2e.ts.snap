// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-notification-toast #render should render alert notification toast 1`] = `"<gux-notification-toast lang=\\"en\\" accent=\\"alert\\" class=\\"hydrated\\"><!----> <div class=\\"gux-icon gux-alert\\"><gux-icon slot=\\"icon\\" icon-name=\\"user-add\\" decorative=\\"\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M6.247 9a3.5 3.5 0 10-3.5-3.5 3.5 3.5 0 003.5 3.5zm2.766 5.084v-1.167a1.166 1.166 0 011.166-1.167h.584v-.822A3.407 3.407 0 008.1 9.6a4.479 4.479 0 01-3.709 0A3.422 3.422 0 001 13.2V15h8.49a1.143 1.143 0 01-.477-.916zm4.125-1.209V11h-1.25v1.875h-1.875v1.25h1.875V16h1.25v-1.875h1.875v-1.25z\\" data-name=\\"add user\\"></path></svg></div></gux-icon></div><div class=\\"gux-content\\"><div class=\\"gux-title\\"><div slot=\\"title\\">Title</div></div><div class=\\"gux-message\\"><div slot=\\"message\\">This is the message</div></div></div><button class=\\"gux-dismiss-button\\" title=\\"Dismiss\\"><gux-icon icon-name=\\"close\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"><path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path></svg></div></gux-icon></button></gux-notification-toast>"`;

exports[`gux-notification-toast #render should render neutral notification toast 1`] = `"<gux-notification-toast lang=\\"en\\" accent=\\"neutral\\" class=\\"hydrated\\"><!----> <div class=\\"gux-icon gux-neutral\\"><gux-icon slot=\\"icon\\" icon-name=\\"user-add\\" decorative=\\"\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M6.247 9a3.5 3.5 0 10-3.5-3.5 3.5 3.5 0 003.5 3.5zm2.766 5.084v-1.167a1.166 1.166 0 011.166-1.167h.584v-.822A3.407 3.407 0 008.1 9.6a4.479 4.479 0 01-3.709 0A3.422 3.422 0 001 13.2V15h8.49a1.143 1.143 0 01-.477-.916zm4.125-1.209V11h-1.25v1.875h-1.875v1.25h1.875V16h1.25v-1.875h1.875v-1.25z\\" data-name=\\"add user\\"></path></svg></div></gux-icon></div><div class=\\"gux-content\\"><div class=\\"gux-title\\"><div slot=\\"title\\">Title</div></div><div class=\\"gux-message\\"><div slot=\\"message\\">This is the message</div></div></div><button class=\\"gux-dismiss-button\\" title=\\"Dismiss\\"><gux-icon icon-name=\\"close\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"><path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path></svg></div></gux-icon></button></gux-notification-toast>"`;

exports[`gux-notification-toast #render should render positive notification toast 1`] = `"<gux-notification-toast lang=\\"en\\" accent=\\"positive\\" class=\\"hydrated\\"><!----> <div class=\\"gux-icon gux-positive\\"><gux-icon slot=\\"icon\\" icon-name=\\"user-add\\" decorative=\\"\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M6.247 9a3.5 3.5 0 10-3.5-3.5 3.5 3.5 0 003.5 3.5zm2.766 5.084v-1.167a1.166 1.166 0 011.166-1.167h.584v-.822A3.407 3.407 0 008.1 9.6a4.479 4.479 0 01-3.709 0A3.422 3.422 0 001 13.2V15h8.49a1.143 1.143 0 01-.477-.916zm4.125-1.209V11h-1.25v1.875h-1.875v1.25h1.875V16h1.25v-1.875h1.875v-1.25z\\" data-name=\\"add user\\"></path></svg></div></gux-icon></div><div class=\\"gux-content\\"><div class=\\"gux-title\\"><div slot=\\"title\\">Title</div></div><div class=\\"gux-message\\"><div slot=\\"message\\">This is the message</div></div></div><button class=\\"gux-dismiss-button\\" title=\\"Dismiss\\"><gux-icon icon-name=\\"close\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"><path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path></svg></div></gux-icon></button></gux-notification-toast>"`;

exports[`gux-notification-toast #render should render warning notification toast 1`] = `"<gux-notification-toast lang=\\"en\\" accent=\\"warning\\" class=\\"hydrated\\"><!----> <div class=\\"gux-icon gux-warning\\"><gux-icon slot=\\"icon\\" icon-name=\\"user-add\\" decorative=\\"\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-hidden=\\"true\\"><path d=\\"M6.247 9a3.5 3.5 0 10-3.5-3.5 3.5 3.5 0 003.5 3.5zm2.766 5.084v-1.167a1.166 1.166 0 011.166-1.167h.584v-.822A3.407 3.407 0 008.1 9.6a4.479 4.479 0 01-3.709 0A3.422 3.422 0 001 13.2V15h8.49a1.143 1.143 0 01-.477-.916zm4.125-1.209V11h-1.25v1.875h-1.875v1.25h1.875V16h1.25v-1.875h1.875v-1.25z\\" data-name=\\"add user\\"></path></svg></div></gux-icon></div><div class=\\"gux-content\\"><div class=\\"gux-title\\"><div slot=\\"title\\">Title</div></div><div class=\\"gux-message\\"><div slot=\\"message\\">This is the message</div></div></div><button class=\\"gux-dismiss-button\\" title=\\"Dismiss\\"><gux-icon icon-name=\\"close\\" class=\\"hydrated\\"><div class=\\"gux-icon-container\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" aria-label=\\"Dismiss\\"><path d=\\"M12.95 4.465L9.414 8l3.535 3.535-1.414 1.415L8 9.415 4.465 12.95l-1.414-1.415L6.586 8 3.051 4.465l1.414-1.414L8 6.586l3.535-3.535 1.415 1.414z\\"></path></svg></div></gux-icon></button></gux-notification-toast>"`;
